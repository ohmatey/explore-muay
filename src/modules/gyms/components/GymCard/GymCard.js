import React from 'react'
import propTypes from 'prop-types'

const GymCard = ({
  title,
  subtitle,
  handleClick
}) => (
  <div>
    <h1>{title}</h1>
    <h2>{subtitle}</h2>

    <button onClick={handleClick}>Go</button>
  </div>
)

GymCard.propTypes = {
  title: propTypes.string
}

export default GymCard
