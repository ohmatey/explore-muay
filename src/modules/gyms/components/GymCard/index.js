import GymCard from './GymCard'

export default GymCard
