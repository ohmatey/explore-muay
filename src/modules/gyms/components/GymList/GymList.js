import React from 'react'

import GymCard from '../GymCard'

const GymList = ({
  gyms = []
}) => (
  <div>
    {gyms.map(gym => (
      <GymCard
        key={gym.title}
        title={gym.title}
        subtitle={gym.subtitle}
      />
    ))}
  </div>
)

export default GymList
