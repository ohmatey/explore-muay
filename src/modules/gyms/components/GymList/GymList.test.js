import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import GymList from './GymList'
import GymCard from '../GymCard'

describe('<GymList />', () => {
  let component

  beforeEach(() => {
    component = shallow(<GymList />)
  })

  it('should render with no errors', () => {
    const tree = toJson(component)

    expect(tree).toMatchSnapshot()
  })

  it('should render a list of GymCards', () => {
    const gyms = [{
      title: 'Bobs Muay Thai',
      subtitle: 'Bob'
    },
    {
      title: 'Sitmonchai',
      subtitle: 'Pee A'
    }]

    component.setProps({
      gyms
    })

    expect(component.find(GymCard).length).toEqual(2)
    expect(component.find(GymCard).at(0).props()).toEqual(gyms[0])
  })
})
