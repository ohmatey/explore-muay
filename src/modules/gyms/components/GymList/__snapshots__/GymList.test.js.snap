// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GymList /> should render with no errors 1`] = `<div />`;
